src/WebJobs.Extensions.CosmosDB/Bindings/CosmosDBAsyncCollector.cs

```diff
@@ -63,18 +56,16 @@ public Task FlushAsync(CancellationToken cancellationToken = default(Cancellatio
             return Task.FromResult(0);
         }
 
-        internal static async Task UpsertDocument(CosmosDBContext context, T item)
+        internal static Task UpsertDocument(CosmosDBContext context, T item)
         {
-            Uri collectionUri = UriFactory.CreateDocumentCollectionUri(context.ResolvedAttribute.DatabaseName, context.ResolvedAttribute.CollectionName);
-
-            // DocumentClient does not accept strings directly.
-            object convertedItem = item;
+            // Support user sending a string
             if (item is string)
             {
-                convertedItem = JObject.Parse(item.ToString());
+                JObject asJObject = JObject.Parse(item.ToString());
```

Comment:
Can we use System.Text.Json here? Would that allow you to remove a ref to Newtonsoft?

Reply:
Nevermind... I forgot we require Newtonsoft.

Reply:
The SDK could take System.Text.Json using a custom serializer if that is the approach we want to take.